components/provider_ppucd/src/parse_ppucd.rs

```diff
@@ -48,33 +52,102 @@ fn is_property_line(line: &str) -> bool {
     let prop_type = &line_parts[1];
     if prop_type == &"Binary" {
-        line_parts.drain(0..2);
-
-        // TODO: ask Markus what to do with the property lines that appear to have
-        // no canonical name
-        // property;Binary;;alnum
-        // property;Binary;;blank
-        // property;Binary;;graph
-        // property;Binary;;print
-        // property;Binary;;xdigit
-        if line_parts[0].is_empty() {
-            line_parts.drain(0..1);
-        }
+        update_aliases(binary_prop_aliases, line);
+    } else if prop_type == &"Enumerated" {
+        update_aliases(enum_prop_aliases, line);
+    }
+}
 
-        let canonical_name = line_parts[0];
-        let all_names: Vec<&'s str> = line_parts.iter().copied().collect();
-        let all_names_set: HashSet<&'s str> = all_names.into_iter().collect();
-        prop_aliases.insert(canonical_name, all_names_set);
+/// Parse enum property value definition line.
+fn update_enum_val_aliases<'s>(
+    enum_val_aliases: &mut HashMap<&'s str, HashMap<&'s str, HashSet<&'s str>>>,
+    line: &'s str,
+) {
+    let line_parts = split_line(&line);
+    let mut line_parts: &[&str] = line_parts.as_slice();
+    assert_eq!(&"value", &line_parts[0]);
+    line_parts = &line_parts[1..];
+    let enum_prop_name = line_parts[0];
+    let enum_prop_val = line_parts[1];
+    enum_val_aliases
+        .entry(enum_prop_name)
+        .or_insert_with(HashMap::new);
+    let enum_val_alias_map: &mut HashMap<&str, HashSet<&str>> =
+        enum_val_aliases.get_mut(&enum_prop_name).unwrap();
+    enum_val_alias_map
+        .entry(enum_prop_val)
+        .or_insert_with(HashSet::new);
+    let enum_prop_val_aliases: &mut HashSet<&str> =
+        enum_val_alias_map.get_mut(&enum_prop_val).unwrap();
+    enum_prop_val_aliases.insert(enum_prop_val);
+    line_parts = &line_parts[2..];
+    // What remains of line_parts are all of the remaining aliases for this
+    // enumerated property's value
+    for alias in line_parts {
+        enum_prop_val_aliases.insert(alias);
     }
 }
 
+/// Return a new map in which any binary exclusion values
+/// (in other words, values for binary properties that are prefixed with a
+/// minus sign, as described in the PPUCD documentation, ex: "-Gr_Base")
+/// are not included and any existing include values for the binary property
+/// (ex: "Gr_Base") are accordingly also removed from the map.
+fn apply_exclude_vals_for_binary_props<'s>(
+    prop_vals: HashMap<&'s str, &'s str>,
+) -> HashMap<&'s str, &'s str> {
+    let mut prop_names: HashSet<&'s str> = prop_vals.keys().copied().collect();
+    // If we see "-Gr_Base", then remove both "Gr_Base" and "-Gr_Base".
+    for prop_name in prop_vals.keys() {
+        if let Some(unadorned_prop_name) = prop_name.strip_prefix('-') {
+            let orig_prop_name: &str = unadorned_prop_name;
+            prop_names.remove(&orig_prop_name);
+            prop_names.remove(prop_name);
+        }
```

Comment:
Drive-by: 

Having both `unadored_prop_name` and `orig_prop_name` seems unnecessary. 

Using just one of them should work fine:
```rust
    for prop_name in prop_vals.keys() {
        if let Some(orig_prop_name) = prop_name.strip_prefix('-') {
            prop_names.remove(orig_prop_name);
            prop_names.remove(prop_name);
        }
    }
```

Reply:
Done.